### ISSUE TITLE
RotatE in FB15K-237

### ISSUE THREAD

**Issue by StrugglerDream** [uma-pi1/kge#195]
Thanks for your work. I'm using it for my homework. But when I using RotatE on FB15K-237, loss of the model didn't descend, performance can't improve, but other model can run effectively, i think there is something wrong about my RotatE yaml config file, but i can‘t find it. Could you please offer an effective example of RotatE yaml file on  FB15K-237? Then I can compare it with my RotatE yaml file and find the error. No need to be the best, just effective is ok, thank you.
   Looking forword to your reply.

**Comment by samuelbroscheit** [comment]
Can you post your current configuration so we can reproduce what you describe?

**Comment by StrugglerDream** [comment]
ok, this is my current configuration:

`job.type: train
dataset.name: fb15k-237

model: rotate

train:
  type: negative_sampling
  batch_size: 1024
  loss: margin_ranking
  loss_arg: 1.0
  max_epochs: 1000
  optimizer.default:
    type: SGD
    args:
      lr: 0.01

valid:
  early_stopping.patience: 3
  every: 10
  metric: mean_reciprocal_rank_filtered_with_test

eval:
  batch_size: 128

negative_sampling:
  num_samples.s: 5  # default: p=0; o=s
  sampling_type: frequency

lookup_embedder:
  dim: 300`

**Comment by AdrianKs** [comment]
Training with pure SGD is quite unstable with KGE in my experience. You could try using either Adagrad or Adam. Additionally your number of negatives is rather low and you are sampling only for the subject slot. Usually we sample for both subject and object. In general you might have to run a hyper-parameter search to find a good setting. You can have a look at how we performed our hyper-parameter search for all models [here](https://github.com/uma-pi1/kge-iclr20).

**Comment by StrugglerDream** [comment]
ok, I will try, thank you sir.

**Comment by rgemulla** [comment]
We may also have a good configuration for this setup. @rufex2001 Do we?
